PWGMM/Rivet/Tasks/rivet.cxx

```diff
@@ -0,0 +1,462 @@
+    // Begin merging of rivet tasks
+    using DeviceSpec = o2::framework::DeviceSpec;
+    using RunningWorkflowInfo = o2::framework::RunningWorkflowInfo;
+
+    // Get other workflows
+    auto& workflows = initContext.services().get<RunningWorkflowInfo const>();
+
+    std::map<std::string, DeviceSpec*> rivets;
+    // Loop over workflow devices
+    for (const DeviceSpec& device : workflows.devices) {
+      auto suf = get_device_name_suffix(device.name);
+
+      // Check if workflow device name matches this
+      if (suf.empty()) {
+        continue;
+      }
+      rivets[suf] = const_cast<DeviceSpec*>(&device);
+    }
+
+    if (rivets.size() <= 1) {
+      // Only one workflow, return immediately - nothing to do
+      return;
+    }
+
+    // get own suffix. Is there a better way?
+    auto& deviceSpec = initContext.services().get<DeviceSpec const>();
+    // The suffix of this device.
+    std::string suffix = get_device_name_suffix(deviceSpec.name);
+
+    // Check if we're the first workflow.  Note, since we use a sorted
+    // map, we can be confidient that the sorting will be the same in
+    // all workflow programs.
+    bool first = rivets.begin()->first == suffix;
+    bool zombie = !first; // Why can I not use the keyword `not` - so
+                          // darn stupid!
+
+    if (zombie) {
+      // Zero the analysis
+      mWrapper.configs.anas.value = "";
+      return;
+    }
+
+    // Current log level, may be -1 if not set
+    int logLevel = mWrapper.findLogLevel(mWrapper.configs.log);
+
+    // Loop over other device specs and check for consistent values of
+    // options.  Note that we cannot change the settings of the other
+    // devices here - they live in a different process and changes
+    // done in this process are not propagated there.  Hence, we have
+    // to fail hard if some setting isn't consistent.
+    for (auto sufdevice : rivets) {
+      std::string suf = sufdevice.first;
+      auto& dev = *sufdevice.second;
+
+      bool self = suf == suffix;
+      if (self) {
+        continue;
+      }
+
+      // Loop over the defined options
+      for (auto& option : dev.options) {
+        // LOG(info) << "  option " << std::quoted(option.name);
+        // Check if we have cross-section
+        if (option.name == mWrapper.configs.crossSection.name) {
+          // Get the value specified, if any
+          auto val = option.defaultValue.get<double>();
+          if (val > 0) {
+            // Why can I not use the keyword `not` - so darn stupid!
+            if (!isclose(val, mWrapper.configs.crossSection.value)) {
+              LOG(fatal) << "Inconsistent cross-section settings for Rivet: "
+                         << val << " versus "
+                         << mWrapper.configs.crossSection;
+            } else {
+              mWrapper.configs.crossSection.value = val;
+            }
+          }
+        } // Do not mess with the formatting
+        // Check if we have merge equivalent
```

Comment:
It's not needed (anymore), since I took out the `else` parts.   Again, I would strongly argue that 

    if (...) {
      ...
    }
    else {
      ...
    }

is _much_ more readable than 

    if (...) {
      ...
    } else {
      ...
    }

where it is easy to miss the `} else {` line.   Also, as long as the code is not horribly formatted (or not formatted at all), we shouldn't really dictate how people write code - it will only serve dissuade people from contributing.